main.tf

```diff
@@ -0,0 +1,148 @@
+/**
+ * Creates a lambda function with associated role and policies, which
+ * will log to Cloudwatch Logs.
+ *
+ * Creates the following resources:
+ *
+ * * Lambda function
+ * * IAM role and policies described in the inputs
+ * * Cloudwatch Logs group
+ *
+ * ## Usage
+ *
+ * ```hcl
+ * module "my_lambda_function" {
+ *   source                = "trussworks/terraform-aws-lambda"
+ *   name                  = "my_app"
+ *   job_identifier        = "instance_alpha"
+ *   runtime               = "go1.x"
+ *   lambda_role_policy    = "${data.aws_iam_policy_document.my_app_policy.json}"
+ *   s3_bucket             = "my_s3_bucket"
+ *   s3_key                = "my_app/1.0/my_app.zip"
+ *
+ *   subnet_ids            = ["subnet-0123456789abcdef0"]
+ *   security_group_ids    = ["sg-0123456789abcdef0"]
+ *
+ *   source_types          = ["events"]
+ *   source_arns           = ["${aws_cloudwatch_event_rule.trigger.arn}"]
+ *
+ *   env_vars {
+ *     VARNAME = "value"
+ *   }
+ *
+ *   tags {
+ *     "Name" = "my_app_lambda"
+ *   }
+ *
+ * }
+ * ```
+ */
+
+data "aws_region" "current" {}
+data "aws_caller_identity" "current" {}
+
+# This is the IAM policy for letting lambda assume roles.
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+# Create the IAM role for the Lambda instance.
+resource "aws_iam_role" "main" {
+  name               = "lambda-${var.name}-${var.job_identifier}"
```

Comment:
This string gets used a lot. I think it should be a `local`.

Reply:
That works. I'll rewrite to incorporate that.